src/components/hv-picker-field/index.js

```diff
@@ -140,24 +148,23 @@ export default class HvPickerField extends PureComponent<
    * Hides the picker and applies the chosen value to the field.
    */
   onModalDone = () => {
-    const element: Element = this.props.element;
+    const newElement: Element = this.element.cloneNode(true);
+    newElement.setAttribute('value', this.state.pickerValue || '');
+    this.element = newElement;
     this.setState({
       focused: false,
-      value: this.state.pickerValue,
     });
-    element.setAttribute('value', this.state.pickerValue || '');
   };
 
   /**
    * Renders the picker component. Picker items come from the
    * <picker-item> elements in the <picker-field> element.
    */
   renderPicker = (style: StyleSheetType<*>): ReactNode => {
-    const element: Element = this.props.element;
+    const element: Element = this.element;
     const props = {
       onValueChange: value => {
         this.setState({ pickerValue: value });
-        element.setAttribute('value', value || '');
       },
```

Comment:
@adamstep @flochtililoch removing this line is enough for the fix, rest are just state cleanups